### PR TITLE
Add numpydoc Pre-Commit For Ultra & Swapi Pt.3

### PR DESCRIPTION
**Pull request by daralynnrhode**
# Add numpydoc Pre-Commit Hook for files contained within ultra and swapi instrument folder.

## Overview
Adding numpydoc pre-commit. Will be doing this in parts to make it more manageable.  Making any needed changes flagged by the new pre-commit ONLY in files listed within `imap_processing/ultra` and `imap_processing/swapi`.

This PR will have changes made in `imap_processing/ultra` and `imap_processing/swapi`.
The folders and files listed above should be the only ones that numpydoc pre-commit checks will be run on.
For the most part, almost all the docstrings contained in the given files had to be updated. Most were small changes, others I had to fill in details based on what I could discern.

The main portions that need checked--and most are marked with TODO Double Check or something similar beside them-- are the Parameters and Returns sections. These are the sections I had to do the most additions in.
## Updated Files

-pyproject.toml
-.pre-commit-config.yaml

ultra files
-imap_processing/ultra/utils/ultra_l1_utils.py
-imap_processing/ultra/l1c/ultra_l1c.py
-imap_processing/ultra/l1c/pset.py
-imap_processing/ultra/l1c/histogram.py
-imap_processing/ultra/l1b/ultra_l1b.py
-imap_processing/ultra/l1b/extendedspin.py
-imap_processing/ultra/l1b/de.py
-imap_processing/ultra/l1b/cullingmask.py
-imap_processing/ultra/l1b/badtimes.py
-imap_processing/ultra/l1a/ultra_l1a.py
-imap_processing/ultra/l0/ultra_utils.py
-imap_processing/ultra/l0/decom_ultra.py
-imap_processing/ultra/l0/decom_tools.py

Swapi files
-imap_processing/swe/utils/swe_utils.py
-imap_processing/swe/l1b/swe_l1b_science.py
-imap_processing/swe/l1b/swe_l1b.py
-imap_processing/swe/l1a/swe_science.py
-imap_processing/swe/l1a/swe_l1a.py
-imap_processing/swe/l0/decom_swe.py
## Testing
-Added numpydoc pre-commits to .pre-commit-config.yaml and pyproject.toml. For now, .pre-commit-config.yaml has exclusions included in it that will be periodically removed as progressed is made in updating all files.

### CODE REVIEW

**Review by laspsandoval — APPROVED** [review]
Nice formatting. Thanks for doing this!

File: imap_processing/swapi/l1/swapi_l1.py [hunk 1]
```diff
@@ -408,14 +419,20 @@ def process_sweep_data(full_sweep_sci, cem_prefix):
...
-    """Process SWAPI science data and create CDF file.
+    """
+    Will process SWAPI science data and create CDF file.
 
     Parameters
     ----------
-    dataset : xarray.Dataset
-        L0 data
+    sci_dataset : xarray.Dataset
+        L0 data. todo check.
```
Comment (maxinelasp) [inline imap_processing/swapi/l1/swapi_l1.py]: Not sure what you're checking, but it is L0 data so this looks good

File: imap_processing/ultra/l1a/ultra_l1a.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-"""Contains code to perform ULTRA L1a cdf generation."""
+"""Contain code to perform ULTRA L1a cdf generation."""
```
Comment (maxinelasp) [inline imap_processing/ultra/l1a/ultra_l1a.py]:
```suggestion
"""Generate ULTRA L1a cdf."""
```

**Review by maxinelasp — APPROVED** [review]
Nice, LGTM! Thanks for doing this!

File: imap_processing/swapi/swapi_utils.py [hunk 3]
```diff
@@ -8,11 +9,13 @@
...
-    """Create ENUM for apid.
+    """
+    Create ENUM for apid.
 
-    Parameters
+    Attributes
     ----------
-    IntEnum : IntEnum.
+    IntEnum : IntEnum
+        No idea todo Check.
```
Comment (bourque) [inline imap_processing/swapi/swapi_utils.py]:
```suggestion
        SWAPI APID Enums
```

File: imap_processing/swapi/swapi_utils.py [hunk 4]
```diff
@@ -21,11 +24,13 @@ class SWAPIAPID(IntEnum):
...
-    """Create ENUM for MODE.
+    """
+    Create ENUM for MODE.
 
-    Parameters
+    Attributes
     ----------
-    IntEnum : IntEnum.
+    IntEnum : IntEnum
+        No idea todo check.
```
Comment (bourque) [inline imap_processing/swapi/swapi_utils.py]:
```suggestion
        SWAPI Mode Enums
```

File: imap_processing/ultra/l1a/ultra_l1a.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-"""Contains code to perform ULTRA L1a cdf generation."""
+"""Generate ULTRA L1a cdf generation."""
```
Comment (bourque) [inline imap_processing/ultra/l1a/ultra_l1a.py]:
```suggestion
"""Generate ULTRA L1a CDFs."""
```

**Review by bourque — APPROVED** [review]
Just a few minor suggestions, otherwise looks good!

**Pull request merged**